src/etcd.js

```diff
@@ -13,15 +14,22 @@ function applyChange (config, change) {
   let changed = false
   const [key, name, group] = getKey(config.prefix, change.node.key)
   const value = change.node.value
-  const level = group ? 4 : (name ? 3 : 2)
+  let level = 2
+  if (group) {
+    level = 4
+  } else if (name) {
+    level = 3
+  }
+
   config.sets.forEach(definition => {
     if (definition.key === key) {
       match = true
       if (level >= definition.level) {
         if (change.action === 'set') {
           definition.setValue(value, level)
-        } else if (change.action === 'delete') {
-          definition.clearValue(value, level)
+        } else {
+          assert(change.action === 'delete')
```

Comment:
Just to be clear we are intentionally changing the behavior here to throw an exception if change.action isn't set or delete?

Reply:
I couldn't see a way to get to this line of code if the action wasn't set to `delete` - I can take another look to make sure I wasn't going crazy though!

Reply:
OK - I can see absolutely no way for this to be anything other than `delete` given the code. However operation will still be the same by leaving the branch in - so I've reverted and added a comment in the code about coverage.